flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -957,4 +969,65 @@ public void run() {
 			}
 		});
 	}
+
+	// ------------------------------------------------------------------------
+	//  pending checkpoints stats metrics
+	// ------------------------------------------------------------------------
+	private class PendingCheckpointStatGauge implements Gauge<List<PendingCheckpointStats>> {
```

Comment:
Can this be refactored into `org.apache.flink.runtime.metrics.util.MetricUtils`?